server/api_server/api_server_start.py

```
@@ -1947,6 +1955,111 @@ def check_auth(payload=None):
+                else:
+                    # Reset after lockout expires
+                    FAILED_LOGINS[client_ip] = (0, now)
+
+    _auth_manager = AuthManager()
+    result = _auth_manager.authenticate(username, password)
+
+    if result.success:
+        # Clear failures on success
+        with _failed_logins_lock:
+            if client_ip in FAILED_LOGINS:
+                del FAILED_LOGINS[client_ip]
```

Ruff (0.15.10)

[warning] Use `pop` instead of `key in dict` followed by `del dict[key]`

Replace `if` statement with `.pop(..., None)`

(RUF051)